src/Jeffijoe.MessageFormat/Formatting/FormatterLibrary.cs

```diff
@@ -45,13 +45,13 @@ public FormatterLibrary()
         /// </exception>
         public IFormatter GetFormatter(FormatterRequest request)
         {
-            var formatter = this.FirstOrDefault(x => x.CanFormat(request));
-            if (formatter == null)
+            foreach (var formatter in this)
             {
-                throw new FormatterNotFoundException(request);
+                if (formatter.CanFormat(request))
+                    return formatter;
             }
 
-            return formatter;
+            throw new FormatterNotFoundException(request);
```

Comment:
Nice! I would imagine this is only slightly faster but it's just as (if not more) readable 👌

Reply:
Yep, the difference in allocations was minor